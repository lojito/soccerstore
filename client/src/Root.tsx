import { type ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { createStore } from "redux";
import reducers from "./reducers";

type Props = {
  children: ReactNode;
  initialStoreValue?: {
    favorites: number[];
    cart: number[];
  };
};

// eslint-disable-next-line react-refresh/only-export-components
export default ({ children, initialStoreValue }: Props) => {
  const store = createStore(
    reducers,
    initialStoreValue as unknown as Partial<{ favorites: never; cart: never }>
  );

  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};
