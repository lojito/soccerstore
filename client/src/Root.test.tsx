import { render } from "@testing-library/react";
import Root from "./Root";

describe("Root", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Root children="children" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
