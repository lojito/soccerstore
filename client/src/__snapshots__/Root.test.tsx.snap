// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root > should render correctly 1`] = `
<DocumentFragment>
  children
</DocumentFragment>
`;
